std/assembly/util/string.ts

```diff
@@ -372,31 +372,34 @@ export function joinFloatArray<T>(dataStart: usize, length: i32, separator: stri
 export function joinStringArray(dataStart: usize, length: i32, separator: string): string {
```

Comment:
Having special code for possibly-null values in `joinReferenceArray` while not also having special code in `joinStringArray` seems a bit inconsistent. My general feeling there is that static null checks are not necessary in `joinReferenceArray` since the operation is expensive enough that a few null checks won't make much of a difference, and it then yields just one instead of two concrete Wasm functions benefiting code size. Now I'm no sure if my feeling there is right, but no matter what we find out, I think both functions should do the same.

Reply:
Another thought: Might even be that `joinReferenceArray` can handle `string` arguments since there's `String#toString` returning the value itself?

Reply:
I favour use different versions for `joinReferenceArray` and `joinStringArray` because last one could be more performant (use prepare total length calc pass and memory.copy)

Reply:
I synced two methods

Reply:
Also wrap `joinStringArray` to `ASC_SHRINK_LEVEL`